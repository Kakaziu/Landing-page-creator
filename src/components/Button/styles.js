import styled, { css } from 'styled-components';

const buttonSizes = {
  small: (theme) => css`
    padding: 8px 20px;
    font-size: ${theme.fonts.sizes.small};
  `,

  big: (theme) => css`
    padding: 8px 30px;
    font-size: ${theme.fonts.sizes.medium};
  `,
};

export const LinkContainer = styled.a``;

export const Button = styled.button`
  ${({ theme, size }) => css`
    font-weight: bold;
    border: none;
    background-color: ${theme.colors.secondaryColor};
    color: ${theme.colors.white};
    border-radius: 4px;
    ${buttonSizes[size](theme)}
  `}
`;
