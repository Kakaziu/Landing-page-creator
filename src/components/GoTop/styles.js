import styled, { css } from 'styled-components';

export const Button = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 4rem;
    right: 4rem;
    width: 40px;
    height: 40px;
    border-radius: 20px;
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: rgba(10, 18, 40, 0.5);
    transition: all 0.3s ease-in-out;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors.primaryColor};
    }
  `}
`;
