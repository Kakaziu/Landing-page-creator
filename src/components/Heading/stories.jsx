import Heading from '.';

export default {
  title: 'Heading',
  component: Heading,
  args: {
    children: 'OlÃ¡ mundo',
    isdark: true,
  },

  argTypes: {
    children: { type: 'string' },
  },

  parameters: {
    background: {
      default: 'light',
    },
  },
};

export const Light = (args) => (
  <div>
    <Heading {...args} />
  </div>
);

export const Dark = (args) => (
  <div>
    <Heading {...args} />
  </div>
);

Dark.parameters = {
  backgrounds: {
    default: 'dark',
  },
};

Dark.args = {
  isdark: false,
};
