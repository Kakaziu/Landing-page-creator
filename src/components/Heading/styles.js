import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,

  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,

  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xxlarge};
  `,

  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

export const Title = styled.h1`
  ${({ theme, isdark, uppercase, size }) => css`
    color: ${isdark ? theme.colors.primaryColor : theme.colors.white};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${titleSizes[size](theme)}
  `}
`;
