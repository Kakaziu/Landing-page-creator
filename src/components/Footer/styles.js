import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    border-top: 1px solid ${theme.colors.mediumGray};
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.small};

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding: ${theme.spacings.small} 0;
    }
  `}
`;
