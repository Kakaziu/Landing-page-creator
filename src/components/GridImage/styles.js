import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { LinkContainer as Button } from '../Button/styles';

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }

    img {
      width: 100%;
      transition: all 300ms ease-in-out;
      cursor: pointer;

      &:hover {
        transform: scale(1.1);
      }
    }
  `}
`;

export const TitleContainer = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.xlarge};
  margin-top: ${theme.spacings.huge};
  width: 100%;

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
  }
`}
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    margin: 8rem 0;
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      color: ${
        background === 1 ? theme.colors.white : theme.colors.primaryColor
      }
    }

    ${Button} {
      margin-top: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;
