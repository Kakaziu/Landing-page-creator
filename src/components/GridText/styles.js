import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const TitleContainer = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.xlarge};
  margin-top: ${theme.spacings.huge};
  width: 100%;

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
  }
`}
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    ${HeadingContainer} {
      margin-bottom: ${theme.spacings.small};
      color: ${
        background === 1 ? theme.colors.white : theme.colors.primaryColor
      }
    }
  `}
`;
