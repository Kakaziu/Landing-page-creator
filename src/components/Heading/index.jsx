import P from 'prop-types';
import * as Styled from './styles';

const Heading = ({
  children,
  isdark = true,
  as = 'h1',
  uppercase = false,
  size = 'medium',
}) => {
  return (
    <Styled.Title isdark={+isdark} as={as} uppercase={+uppercase} size={size}>
      {children}
    </Styled.Title>
  );
};

Heading.propTypes = {
  children: P.node.isRequired,
  isdark: P.bool,
  as: P.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  uppercase: P.bool,
  size: P.oneOf(['small', 'medium', 'big', 'huge']),
};

export default Heading;
