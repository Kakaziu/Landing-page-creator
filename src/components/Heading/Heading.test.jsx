import { screen } from '@testing-library/react';
import Heading from '.';
import renderTheme from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render heading with default values', () => {
    renderTheme(<Heading>Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.fonts.sizes.large,
      'text-transform': 'none',
    });
  });

  it('should content be white when background dark', () => {
    renderTheme(<Heading isdark={false}>Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render with small size', () => {
    renderTheme(<Heading size="small">Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.medium,
    });
  });

  it('should render with big size', () => {
    renderTheme(<Heading size="big">Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xxlarge,
    });
  });

  it('should render with huge size', () => {
    renderTheme(<Heading size="huge">Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xhuge,
    });
  });

  it('should render with uppercase letters', () => {
    renderTheme(<Heading uppercase>Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render with correct heading type', () => {
    const { container } = renderTheme(<Heading as="h6">Oi</Heading>);
    const heading = container.querySelector('h6');

    expect(heading.tagName.toLowerCase()).toBe('h6');
  });

  it('should decrease font size when screen is smaller', () => {
    renderTheme(<Heading size="huge">Oi</Heading>);

    const heading = screen.getByRole('heading', { name: 'Oi' });

    expect(heading).toHaveStyleRule('font-size', theme.fonts.sizes.xlarge, {
      media: theme.media.lteMedium,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading as="h6">texto</Heading>);

    expect(container).toMatchSnapshot();
  });
});
