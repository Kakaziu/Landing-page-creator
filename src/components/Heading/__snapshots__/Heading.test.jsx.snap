// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  color: #0A1128;
  text-transform: none;
  font-size: 3.2rem;
}

<div>
  <h6
    class="c0"
    isdark="1"
    uppercase="0"
  >
    texto
  </h6>
</div>
`;
